### DISCUSSION TITLE
focusAtom is inferring to unknown when moduleResolution is "bundler"

### DISCUSSION THREAD

**Discussion by jonybekov** [pmndrs/jotai#1954] [General]
## Summary
Hi, I'm using focusAtom in React 18, vite 4.3, typescript 5.0.4 web app.
The problem is when I use focusAtom, it doesn't inferring correct types.

![image](https://github.com/pmndrs/jotai/assets/39439363/be37fdc8-967f-48d9-b6ef-8e9cb651bdf1)

If I change `moduleBundler` to "node" in my tsconfig file, it solves the issue. But dayjs import is failing with "node" option. 
I can't understand why focusAtom cannot work properly with "bundler" module resolution...

## Link to reproduction
https://stackblitz.com/edit/vitejs-vite-vsffeg?file=src%2Fatom.ts,tsconfig.json&terminal=dev

## Check List

Please do not ask questions in issues.

- [ ] I've already opened a [discussion](https://github.com/pmndrs/jotai/discussions) before opening this issue, or already discussed in other media.

Please include a minimal reproduction.

- [x] I've added a link to a [typescript playground](https://www.typescriptlang.org/play) or [codesandbox](https://codesandbox.io) with a minimal reproduction.

**Comment by arjunvegda** [comment]
This appears to be an issue with `optics-ts` 😟  
Possibly a duplicate of https://github.com/jotaijs/jotai-optics/issues/6
